Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -74,27 +83,66 @@ class Navigator {
                                                      config: NavigatorConfig(),
                                                      customConfig: "")
         
-        instance.historyRecorder = try! HistoryRecorderHandle.build(forHistoryFile: "",
-                                                                    config: configFactory)
+        historyRecorder = try! HistoryRecorderHandle.build(forHistoryFile: "", config: configFactory)
         
         let runloopExecutor = try! RunLoopExecutorFactory.build()
-        instance.cacheHandle = try! CacheFactory.build(for: tilesConfig,
-                                                       config: configFactory,
-                                                       runLoop: runloopExecutor,
-                                                       historyRecorder: instance.historyRecorder)
+        cacheHandle = try! CacheFactory.build(for: tilesConfig,
+                                              config: configFactory,
+                                              runLoop: runloopExecutor,
+                                              historyRecorder: historyRecorder)
         
-        instance.navigator = try! MapboxNavigationNative.Navigator(config: configFactory,
-                                                                   runLoopExecutor: runloopExecutor,
-                                                                   cache: instance.cacheHandle,
-                                                                   historyRecorder: instance.historyRecorder)
+        roadGraph = RoadGraph(try! MapboxNavigationNative.GraphAccessor(cache: cacheHandle))
         
-        return instance
-    }()
+        navigator = try! MapboxNavigationNative.Navigator(config: configFactory,
+                                                          runLoopExecutor: runloopExecutor,
+                                                          cache: cacheHandle,
+                                                          historyRecorder: historyRecorder)
+        try! navigator.setElectronicHorizonObserverFor(self)
+    }
     
-    /**
-     Restrict direct initializer access.
-     */
-    private init() {
-        
+    deinit {
+        try! navigator.setElectronicHorizonObserverFor(nil)
+    }
+    
+    var electronicHorizonOptions: ElectronicHorizonOptions? {
+        didSet {
+            let nativeOptions: MapboxNavigationNative.ElectronicHorizonOptions?
+            if let electronicHorizonOptions = electronicHorizonOptions {
+                nativeOptions = MapboxNavigationNative.ElectronicHorizonOptions(electronicHorizonOptions)
+            } else {
+                nativeOptions = nil
```

Comment:
NIT: this is redundant. `nativeOptions` will be defaulted to `nil`

Reply:
`nativeOptions` is declared but not initialized as a constant right above here, so the explicit `nil` is required.